### ISSUE TITLE
[feature] Content Security Policy (CSP)

### ISSUE THREAD

**Issue by thealjey** [zeit/styled-jsx#423]
It is possible to specify a `nonce` prop when rendering `<NextScript />`, but there's currently no way to do the same for `styled-jsx` inline styles, forcing me to resort to using `'unsafe-inline'` for styles.

I would like to be able to pass any number of props supported by `<style />` to `'styled-jsx/server'` and have them rendered in the DOM.

I can make a pull request (although, that would require 2 pull requests to both `next` and `styled-jsx`), if you are open to the idea.

**Comment by giuseppeg** [comment]
Definitely open to add this if you prepare a PR.

**Comment by jlecomte** [comment]
Here is a workaround:

```js
let styles = flush()
styles = styles.map(({props, key}) =>
  React.createElement('style', {...props, key, nonce})
)
```

I purposefully did not use JSX to avoid going through the styled-jsx babel plugin, which causes all kinds of strange issues.

**Comment by thealjey** [comment]
Unfortunately, I will most probably not have the time in the next several months to do these pull requests.
Maybe someone else will do them in the meantime 🤞

**Comment by andy-viv** [comment]
@thealjey: the workaround by @jlecomte would be in your code, not in styled-jsx's code.

**Comment by thealjey** [comment]
@andy-viv 
1. I was not referring to his workaround
2. I would rather not have to do any workarounds
3. Next.js does the `flush`, so more workarounds are necessary to make this work

**Comment by giuseppeg** [comment]
fixed in #482
